src/messages.rs

```diff
@@ -135,6 +135,27 @@ pub enum DirectMessage {
     TunnelClosed(PeerId),
     /// Sent to a tunnel node to indicate the tunnel is not needed any more.
     TunnelDisconnect(PeerId),
+    /// Request a proof to be provided by the joining node
+    ///
+    /// This is sent from member of Group Y to the joining node
+    ResourceProof {
+        /// seed of proof
+        seed: Vec<u8>,
+        /// size of the proof
+        target_size: usize,
```

Comment:
I'm never a fan of using variable-sized integers in messages which are passed between machines of differing architectures, but bincode should handle this I suppose...